### PR TITLE
Fix release branch good build detection

### PR DESCRIPTION
**Pull request by sindrig**
This closes island-is/infrastructure#287
Also package lock detection script and run push on release branch
creation like it was supposed to all the time.

### CODE REVIEW

File: .github/actions/main.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ rl.on('line', function (line) {
 rl.on('close', async function () {
   const result = await findLastGoodBuild(
     shas,
-    process.env.BRANCH,
+    process.env.BRANCH || process.env.GIT_BRANCH,
```
Comment (sindrig) [inline .github/actions/main.ts:41]: This is basically a case of "Computer, do what I want you to do, not what I tell you to do."

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -45,13 +45,14 @@ jobs:
...
                   echo "'$GIT_BRANCH' matches '$pattern', continuing with build"
+                  echo "::set-output name=PRE_CHECK_OK::$GITHUB_EVENT_NAME"
                   exit 0
                 fi
             done
             echo "No pattern matches '$GIT_BRANCH', exiting with exit code 1"
             exit 1
           fi
-          echo "::set-output name=PRE_CHECK_OK::true"
+          echo "::set-output name=PRE_CHECK_OK::$GITHUB_EVENT_NAME"
```
Comment (pshomov) [inline .github/workflows/push.yml:55]: Not quite sure what's the improvement here? Thought that "true" was clearer if anything. Could not find clear rules about typecasting from string => bool but I guess JS rules apply.

Comment (sindrig) [reply .github/workflows/push.yml:55]: Not really an improvement, just some leftover from when I intended to use exit code for flow control.

I felt it was better to have some distinction between output values depending on if it was set by a matching branch or not-create. 

I'm pretty sure we're dealing with js truthy rules, so empty string/null vs any string contents.

Comment (pshomov) [reply .github/workflows/push.yml:55]: 👍🏻

**Review by pshomov — APPROVED** [review]
LGTM

**Pull request merged**